phoenix-core/src/it/java/org/apache/phoenix/end2end/index/GlobalIndexCheckerIT.java

```diff
@@ -0,0 +1,231 @@
+                dataTableName + " (val1) include (val2, val3)" + (async ? "ASYNC" : ""));
+        if (async) {
+            // run the index MR job.
+            IndexToolIT.runIndexTool(true, false, null, dataTableName, indexName);
+        }
+        // Configure Indexer to skip the last write phase (i.e., the post index update phase) where the verify flag is set
+        // to true and/or index rows are deleted and check that this does not impact the correctness
+        Indexer.setSkipPostIndexUpdatesForTesting(true);
+        conn.createStatement().execute("upsert into " + dataTableName + " (id, val2) values ('a', 'abcc')");
+        conn.commit();
+        conn.createStatement().execute("upsert into " + dataTableName + " (id, val1, val2) values ('c', 'cd','cde')");
+        conn.commit();
+        String selectSql =  "SELECT val2, val3 from " + dataTableName + " WHERE val1  = 'ab'";
+        // Verify that we will read from the index table
+        assertExplainPlan(conn, selectSql, dataTableName, indexName);
+        ResultSet rs = conn.createStatement().executeQuery(selectSql);
+        assertTrue(rs.next());
+        assertEquals("abcc", rs.getString(1));
+        assertEquals("abcd", rs.getString(2));
+        assertFalse(rs.next());
+        conn.close();
+    }
+
+    @Test
+    public void skipDataTableAndPostIndexPartialRowUpdate() throws Exception {
+        String dataTableName = generateUniqueName();
+        populateTable(dataTableName);
+        Connection conn = DriverManager.getConnection(getUrl());
+        String indexName = generateUniqueName();
+        conn.createStatement().execute("CREATE INDEX " + indexName + " on " +
+                dataTableName + " (val1) include (val2, val3)" + (async ? "ASYNC" : ""));
+        if (async) {
+            // run the index MR job.
+            IndexToolIT.runIndexTool(true, false, null, dataTableName, indexName);
+        }
+        // Configure Indexer to skip the last two write phase (i.e., the data table update and post index update phase)
+        // and check that this does not impact the correctness
+        Indexer.setSkipDataTableUpdatesForTesting(true);
+        Indexer.setSkipPostIndexUpdatesForTesting(true);
+        conn.createStatement().execute("upsert into " + dataTableName + " (id, val2) values ('a', 'abcc')");
+        conn.commit();
+        Indexer.setSkipDataTableUpdatesForTesting(false);
+        Indexer.setSkipPostIndexUpdatesForTesting(false);
+        conn.createStatement().execute("upsert into " + dataTableName + " (id, val3) values ('a', 'abcdd')");
+        conn.commit();
+        String selectSql =  "SELECT val2, val3 from " + dataTableName + " WHERE val1  = 'ab'";
+        // Verify that we will read from the index table
+        assertExplainPlan(conn, selectSql, dataTableName, indexName);
+        ResultSet rs = conn.createStatement().executeQuery(selectSql);
+        assertTrue(rs.next());
+        assertEquals("abc", rs.getString(1));
```

Comment:
should not this value be abcc? It is reading the old value.

Reply:
Note the select statement is "SELECT val2, val3 from ".  So, I am reading the columns val2 and val3. The last two upserts were
"upsert into " + dataTableName + " (id, val2) values ('a', 'abcc')")
 "upsert into " + dataTableName + " (id, val3) values ('a', 'abcdd')")
So, val2 = 'abcc' and val3 = 'abcdd'

Reply:
Sorry, I still did not understand. if val2 is abcc, why is assertEquals not abcc? It is passing with abc

Reply:
Sorry, I was thinking  that we were talking about skipping only post index updates (which is the other partial update test). This test also skips data table updates for the first upsert. So the first upsert statement (i.e., "upsert into " + dataTableName + " (id, val2) values ('a', 'abcc')") should have no effect. Thus, val2 stays as "abc"